hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/ozone/container/common/TestEndPoint.java

```diff
@@ -178,32 +187,61 @@ public void testTmpDirCleanup() throws Exception {
 
       String clusterId = scmServerImpl.getClusterId();
 
-      // write some data before calling versionTask.call()
+      // VolumeSet for this test, contains only 1 volume
       MutableVolumeSet volumeSet = ozoneContainer.getVolumeSet();
-      for (HddsVolume hddsVolume : StorageVolumeUtil.getHddsVolumesList(
-          volumeSet.getVolumesList())) {
-        StorageVolumeUtil.checkVolume(hddsVolume, clusterId,
-            clusterId, conf, null, null);
-
-        // Create a container and move it under the tmp delete dir.
-        KeyValueContainer container = ContainerTestUtils.
-            setUpTestContainerUnderTmpDir(hddsVolume, clusterId,
-                conf, OzoneConsts.SCHEMA_V3);
-        Assertions.assertTrue(container.getContainerFile().exists());
-      }
+      Assertions.assertEquals(1, volumeSet.getVolumesList().size());
+      StorageVolume volume = volumeSet.getVolumesList().get(0);
 
-      VersionEndpointTask versionTask = new VersionEndpointTask(rpcEndPoint,
-          conf, ozoneContainer);
-      EndpointStateMachine.EndPointStates newState = versionTask.call();
+      // Check instanceof and typecast
+      Assertions.assertTrue(volume instanceof HddsVolume);
+      HddsVolume hddsVolume = (HddsVolume) volume;
 
-      Assertions.assertEquals(EndpointStateMachine.EndPointStates.REGISTER,
-          newState);
+      StorageVolumeUtil.checkVolume(hddsVolume, clusterId,
```

Comment:
We should re-use the `ContainerTestUtils.createDbInstancesForTestIfNeeded` to setup test here instead of the `checkVolume`.